index.js

```diff
@@ -29,7 +29,7 @@ exports.parse = function (str) {
 		// http://w3.org/TR/2012/WD-url-20120524/#collect-url-parameters
 		val = val === undefined ? null : decodeURIComponent(val);
 
-		if (!ret.hasOwnProperty(key)) {
+		if (!ret[key]) {
```

Comment:
Thank you for fixing this!  However, I think this conditional is incorrect when the value is null or the empty string.
For example `require('query-string').parse('a&a=&a=2')`